WebServer/frontend/src/app/login/loginDialog.component.ts

```diff
@@ -9,32 +9,32 @@ export interface DialogData {
 }
 
 @Component({
-    selector: 'login-dialog',
-    templateUrl: './loginDialog.component.html',
-  })
-  export class LoginDialog {
-    requesting:boolean = false;
-    failedLogin:boolean = false;
-    constructor(
-      public dialogRef: MatDialogRef<LoginDialog>,
-      @Optional() @Inject(MAT_DIALOG_DATA) public data: DialogData,
-      public loginService: LoginService,
-      ) {}
-  
-    onNoClick(): void {
-      this.dialogRef.close();
-    }
-    async awaitLogin(){
-      this.requesting = true;
-      const res = await (await this.loginService.login(this.data.email, this.data.password));
-      this.requesting = false;
+  selector: 'login-dialog',
+  templateUrl: './loginDialog.component.html',
+})
+export class LoginDialog {
+  requesting: boolean = false;
+  failedLogin: boolean = false;
+  constructor(
+    public dialogRef: MatDialogRef<LoginDialog>,
+    @Optional() @Inject(MAT_DIALOG_DATA) public data: DialogData,
+    public loginService: LoginService,
+  ) { }
+
+  onNoClick(): void {
+    this.dialogRef.close();
+  }
 
-      if(res == 200){
-        this.dialogRef.close(this.data.email);
-      }
-      else{
-        this.failedLogin = true;
-      }
+  async awaitLogin() {
+    this.requesting = true;
+    const res = await (await this.loginService.login(this.data.email, this.data.password));
```

Comment:
Hvorfor er det nødvendigt med to awaits?

Reply:
Ups :P